src/org/jgroups/stack/GossipRouter.java

```diff
@@ -221,6 +216,13 @@ public GossipRouter start() throws Exception {
         server.receiver(this).setMaxLength(max_length)
           .addConnectionListener(this)
           .connExpireTimeout(expiry_time).reaperInterval(reaper_interval).linger(linger_timeout);
+
+        if(diag.isEnabled()) {
```

Comment:
need to be moved one line more, after `server.start()` otherwise `local_addr` will be `null`.